tests/conftest.py

```diff
@@ -995,3 +1000,13 @@ async def have_connections() -> bool:
 @pytest.fixture(name="database_uri")
 def database_uri_fixture() -> str:
     return f"file:db_{random.randint(0, 99999999)}?mode=memory&cache=shared"
+
+
+@pytest.fixture(name="available_listen_port", scope="session")
```

Comment:
`scope="session"` means it will be run once and that same result will be used in every test that needs it.  This doesn't seem good.  Am I missing something?

Reply:
I suppose I should caveat that I'm not sure with pytest-xdist if this ends up being once per session or once per process.  Either way doesn't seem great.

Reply:
oh i forgot about the caching, i was trying to get the recent_ports thing to be shared per session...

Reply:
`recent_ports` is already a global (within the process).

Reply:
> `recent_ports` is already a global (within the process).

i assumed pytest would somehow mess that up but ill set the scope to per function.